theme-utils.mjs

```diff
@@ -1570,6 +1559,283 @@ async function validateThemes( themes, { format, color, tableWidth } ) {
 			);
 		}
 
+		const validators = {
+			validateVersion( attr, value, validLengths = [ 3 ] ) {
+				const problems = [];
+				const adjustedValue =
+					value && `${ value }.0`.split( '.', 3 ).join( '.' );
+				if (
+					! value ||
+					! validLengths.includes( value.split( '.' ).length ) ||
+					! semver.valid( adjustedValue )
+				) {
+					problems.push( {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `format ${ chalkStr.yellow(
+							Array.from( { length: Math.min( validLengths ) } )
+								.fill( 'x' )
+								.join( '.' )
+						) }`,
+						message: `${ value } is not a valid version`,
+					} );
+				}
+				return { isValid: ! problems.length, problems };
+			},
+			validateVersionGte( attr, value, version ) {
+				const problems = [];
+				const adjustedValue =
+					value && `${ value }.0`.split( '.', 3 ).join( '.' );
+				const adjustedVersion =
+					version && `${ version }.0`.split( '.', 3 ).join( '.' );
+				if (
+					! value ||
+					! version ||
+					! semver.valid( adjustedValue ) ||
+					! semver.valid( adjustedVersion ) ||
+					! semver.gte( adjustedValue, adjustedVersion )
+				) {
+					problems.push( {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `${ chalkStr.yellow( version ) } or greater`,
+						message: `provide a valid version value`,
+					} );
+				}
+				return { isValid: ! problems.length, problems };
+			},
+			validateUri: ( attr, value ) => {
+				const problems = [];
+				if ( value && ! URL.canParse( value ) ) {
+					problems.push( {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `a valid URI`,
+						message: `${ value } is not a valid URI`,
+					} );
+				}
+				return { isValid: ! problems.length, problems };
+			},
+			validateThemeSlug: ( attr, value ) => {
+				const problems = [];
+				if ( value && ! /^[a-z0-9-]+$/.test( value ) ) {
+					problems.push( {
+						actual: `${ chalkStr.green(
+							attr
+						) }: ${ chalkStr.yellow( value ) }`,
+						expected: `a valid value`,
+						message: `${ value } is not a valid value`,
+					} );
+				}
+				return { isValid: ! problems.length, problems };
+			},
+			// a8c validations
+			validateA8CThemeUri: ( attr, value ) => {
+				const problems = [];
+				if (
+					value &&
+					! /^https:\/\/wordpress\.com\/themes\/[a-z0-9-]+\/?$/.test(
```

Comment:
It seems the `/themes/{ slug }` URLs redirect to `/theme/{ slug }`, so maybe this should allow both or just `/theme`?

Reply:
Good catch. I think we should allow both. fixed in [75fd4e9](https://github.com/Automattic/themes/pull/8068/commits/75fd4e9f2b2a9ad3802814050e10b41ce372a783)